pose_to_segments/src/model.py

```diff
@@ -67,26 +71,60 @@ def training_step(self, batch, *unused_args):
     def validation_step(self, batch, *unused_args):
         return self.step(batch, *unused_args, name="validation")
 
+    def evaluate(self, level, _gold, _probs, _segments_gold, _mask):
+        metrics = {
+            'loss': [],
+            'frame_accuracy': [],
+            'segment_percentage': [],
+            'segment_IoU': [],
+        }
+
+        for gold, probs, segments_gold, mask in zip(_gold, _probs, _segments_gold, _mask):
+            if level == 'sign':
+                losses = self.sign_loss_function(probs, gold)
+            elif level == 'sentence':
+                losses = self.sentence_loss_function(probs, gold)
+            metrics['loss'].append((losses * mask).mean())
+
+            # assign masked postions the O tag
+            probs_masked = probs.detach().clone()
+            probs_masked[(mask == 0).nonzero(as_tuple=True)[0]] = torch.log(torch.tensor([1, 0, 0]).cuda())
+
+            metrics['frame_accuracy'].append(frame_accuracy(probs_masked, gold))
+
+            segments = probs_to_segments(probs_masked.cpu())
+            # convert segments from second to frame
+            fps = 25
```

Comment:
`fps` is actually a parameter in the data. looking at `batch["pose"]["obj"].body.fps` should give you the number as well